### DISCUSSION TITLE
Get Payment Information

### DISCUSSION THREAD

**Discussion by sarathpm-l7** [SalesforceCommerceCloud/sfcc-ci#437] [Team Posts]
Hi All,

Is there a way to fetch payment information of a customer added in his account. In the customer object, I'm getting the payment information as _'xxxxxxxx1111'_. How can I get the full credit card details? The requirement is to get the payment information from other customers so that they can send money to each other.

Thanks and Regards.

**Comment by rajat-p-talwar** [comment]
Hi,
Check out this [documentation](https://documentation.b2c.commercecloud.salesforce.com/DOC1/index.jsp?topic=%2Fcom.demandware.dochelp%2FDWAPI%2Fscriptapi%2Fhtml%2Fapi%2Fclass_dw_customer_CustomerPaymentInstrument.html&resultof=%22%64%77%2e%63%75%73%74%6f%6d%65%72%2e%43%75%73%74%6f%6d%65%72%50%61%79%6d%65%6e%74%49%6e%73%74%72%75%6d%65%6e%74%22%20) link.
You can use the getCreditCardNumber method to get the decrypted card number. If the mentioned conditions are met, you can retrieve the actual number otherwise it returns the masked number.

**Comment by svarshney123** [comment]
We can't get full card number and  demandware  is  allowing to store last 4
digit of the card Number in profile.

**Comment by sanjaygautampfsweb-zz** [comment]
Hi Guys,

We can decrypt card number, In order to decrypt credit card number.
You should contact to CyberSource team for public and private keys for secure communication.
However, you can use below code as pseudo (Controller version).

===========================================================
var profile : dw.customer.Profile = args.profile;
var instruments : Collection = profile.getWallet().getPaymentInstruments(dw.order.PaymentInstrument.METHOD_CREDIT_CARD);

var certRef : dw.crypto.CertificateRef = new dw.crypto.CertificateRef("server");
var keyRef : dw.crypto.CertificateRef = new dw.crypto.KeyRef("private");

var custPay : PaymentInstrument = null;
var cipher : Cipher = null;
var encryCC = '';
var decCC = '';
var cipher = new Cipher();
if(instruments.size() > 0){
for each (var e : PaymentInstrument in instruments){
       if (e.creditCardNumber != null){
              encryCC = e.getEncryptedCreditCardNumber(dw.order.PaymentInstrument.ENCRYPTION_ALGORITHM_RSA, certRef);
decCC = cipher.decrypt(encryCC,keyRef,dw.order.PaymentInstrument.ENCRYPTION_ALGORITHM_RSA,null,0);             }
       }
}
===========================================================

Hope, it’s helpful.
Let me know if you have any concern.

Thanks
Sanjay
